syncd/Syncd.cpp

```diff
@@ -898,6 +959,122 @@ sai_status_t Syncd::processEntry(
+    std::vector<const sai_attribute_t*> attr_lists(objectIds.size());
+
+    for (size_t idx = 0; idx < objectIds.size(); idx++)
+    {
+        sai_deserialize_object_id(objectIds[idx], objectVids[idx]);
+
+        attr_counts[idx] = attributes[idx]->get_attr_count();
+        attr_lists[idx] = attributes[idx]->get_attr_list();
+    }
+
+    sai_object_id_t switchRid = SAI_NULL_OBJECT_ID;
+
+    if (objectVids.size() > 0 && objectType != SAI_OBJECT_TYPE_SWITCH)
+    {
+        sai_object_id_t switchVid = VidManager::switchIdQuery(objectVids.front());
+        switchRid = m_translator->translateVidToRid(switchVid);
+    }
+
+    std::vector<sai_object_id_t> objectRids(objectIds.size());
+
+    status = m_vendorSai->bulkCreate(objectType, switchRid, (uint32_t)objectIds.size(), attr_counts.data(),
+                                        attr_lists.data(), mode, objectRids.data(), statuses.data());
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        /*
+         * Object was created so new object id was generated we need to save
+         * virtual id's to redis db.
+         */
+        for (size_t idx = 0; idx < objectIds.size(); idx++)
+        {
+            m_translator->insertRidAndVid(objectRids[idx], objectVids[idx]);
+            SWSS_LOG_INFO("saved VID %s to RID %s",
+                    sai_serialize_object_id(objectVids[idx]).c_str(),
+                    sai_serialize_object_id(objectRids[idx]).c_str());
+        }
+    }
+
+    return status;
+}
+
+sai_status_t Syncd::processBulkOidRemove(_In_ sai_object_type_t objectType,
+            _In_ sai_bulk_op_error_mode_t mode,
+            _In_ const std::vector<std::string>& objectIds,
+            _Out_ std::vector<sai_status_t>& statuses)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * Switch, Port objects doesn't have bulk API`s, so we no need additional
+     * functionality for process it
+     */
+    if (objectType == SAI_OBJECT_TYPE_SWITCH || objectType == SAI_OBJECT_TYPE_PORT)
+    {
+        return SAI_STATUS_NOT_SUPPORTED;
+    }
+
+    sai_status_t status;
+
+    std::vector<sai_object_id_t> objectVids(objectIds.size());
+    std::vector<sai_object_id_t> objectRids(objectIds.size());
+
+    for (size_t idx = 0; idx < objectIds.size(); idx++)
+    {
+        sai_deserialize_object_id(objectIds[idx], objectVids[idx]);
+        objectRids[idx] = m_translator->translateVidToRid(objectVids[idx]);
+    }
+
+    status = m_vendorSai->bulkRemove(objectType, (uint32_t)objectIds.size(),
+                                objectVids.data(), mode, statuses.data());
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        /*
+         * remove all related objects from REDIS DB and also from existing
+         * object references since at this point they are no longer valid
+         */
+        sai_object_id_t switchVid;
+        for (size_t idx = 0; idx < objectIds.size(); idx++)
+        {
+            m_translator->eraseRidAndVid(objectRids[idx], objectVids[idx]);
```

Comment:
you need to check each status vector and then if success then execute for contents for that index